### DISCUSSION TITLE
datadome fails

### DISCUSSION THREAD

**Discussion by rechtem** [ultrafunkamsterdam/undetected-chromedriver#1196] [General]
undetected-chromedriver 3.4.6 fails on https://www.leboncoin.fr

**Comment by mir3u** [comment]
Undetected has not worked for datadome for some other sites. What worked for me to pass the datadome system was last version of headless puppeteer+extra packages and a system that saves the datadome cookie from successful requests. It would be nice to be able to make it work into this project, but so far all my attempts were not working.

**Reply by NCLnclNCL** [reply]
I see it too cannot pass checkbox of cloudflare

**Reply by juhacz** [reply]
The site I am using has Datadome protection and UCD is detected if I am using the latest version of UCD. Install version 3.2.1 which works fine and is not detected.

**Reply by rechtem** [reply]
> The site I am using has Datadome protection and UCD is detected if I am using the latest version of UCD. Install version 3.2.1 which works fine and is not detected.

Thanks for the suggestion. I get also detected with 3.2.1 version...

**Reply by mir3u** [reply]
I have been doing tests on datadome with undetected ever since v3 and all of them have been unsuccessful...

**Reply by mir3u** [reply]
> I see it too cannot pass checkbox of cloudflare

That checkbox is actually Turnstile captcha, is something similar to reCaptcha v3 where it runs a javascript script in the background, undetected passes none of them. From experience you need a captcha farm service like 2captcha to pass.

**Reply by rechtem** [reply]
> > I see it too cannot pass checkbox of cloudflare
> 
> That checkbox is actually Turnstile captcha, is something similar to reCaptcha v3 where it runs a javascript script in the background, undetected passes none of them. From experience you need a captcha farm service like 2captcha to pass.

Even filling the capcha by hand, it loops back on the captcha.

**Reply by mir3u** [reply]
yes, it is because even though that checkbox appears, the captcha is still considered failed. 2Captcha extension has implemented a method to override the captcha render function and this is the way to solve it, along their api. You can investigate in the browser by looking for window.turnstile.render, you can catch some ways in which this captcha works

**Reply by rechtem** [reply]
> yes, it is because even though that checkbox appears, the captcha is still considered failed.

Sorry I don't understand, what you mean. When I solve the audio captcha (the puzzle is not working), it says successfully resolved, but I get another captcha.
